test_run.lua

```diff
@@ -40,8 +40,12 @@ local function get_param(self, node, param)
 end
 
 local function get_lsn(self, node, sid)
-    local nodes = self:get_param(node, 'vclock')
-    return tonumber(nodes[1][tonumber(sid)])
+    local node = self:get_param(node, 'vclock')[1]
+    if node == nil then
+        return -1 -- bootstrap in progress
```

Comment:
To be honest, giving -1 here does not look natural for me. When we get a vclock component on a freshly bootstrapped instance (`box.info.vclock[1]`) in a local console, we'll receive `nil`. I would keep the behaviour consistent with tarantool's Lua API.

What does it mean in context of the patch? We should revert the changes in `get_lsn()` and handle `nil` result gracefully in `wait_lsn()`.

Reply:
Used your reproducer to check that tonumber routine removed 'nil' on given '{}'. Added check in wait_lsn routine that get_lsn routine returned not 'nil' value. Checked the fix with test that uses wait_lsn routine:
```
./test-run.py replication/gh-4730-applier-rollback.test.lua
```
with reproducer of the nil value to be sure that the issue fixed:
```
diff --git a/test_run.lua b/test_run.lua
index 7425b4a..eff78dc 100644
--- a/test_run.lua
+++ b/test_run.lua
@@ -68,6 +68,7 @@ local function wait_lsn(self, waiter, master)
     repeat
         fiber.sleep(0.001)
         local waiter_lsn = self:get_lsn(waiter, sid)
+        waiter_lsn = nil
     until waiter_lsn ~= nil and waiter_lsn >= lsn
 end
 
```